src/mon/MgrMonitor.cc

```diff
@@ -517,7 +519,7 @@ bool MgrMonitor::prepare_beacon(MonOpRequestRef op)
       mon.osdmon()->wait_for_writeable(op, new C_RetryMessage(this, op));
       return false;
     }
-    drop_active();
+    plugged |= drop_active();
```

Comment:
Just defensive coding if e.g. we `drop_active` twice for some odd reason in the future. Or, we plug PAXOS in some other method.

Reply:
> > Or, we plug PAXOS in some other method.
> 
> Not following -- wouldn't that single `drop_active` call would just return false in that case?

Yes. If you're talking about an assignment:

`plugged = drop_active(...);`

then the second false would clear `plugged`. Unless you're saying:

```
 plugged = true;
 drop_active(...);
```

I don't like that because it's not clear (via a return value) that the caller is expected to unplug PAXOS.

Reply:
> Yes. If you're talking about an assignment:
> 
> `plugged = drop_active(...);`
> 
> then the second false would clear `plugged`.

Yes, I was talking about this one.  The point is that there is nowhere for the second `false` to come from.  `plugged` is initialized to `false` and then `drop_active` is called, exactly once.

Reply:
That's currently the case, sure. `plugged` may also be set in some other method in the future. The code is structured in a way to indicate there is "state" (PAXOS is plugged by this callstack), that state may be set in some called method, and that state needs cleaned up. I don't think this `|=` should be controversial.